horreum-backend/src/test/java/io/hyperfoil/tools/horreum/svc/RunServiceTest.java

```diff
@@ -77,6 +92,52 @@ public void testDataSetCreated(TestInfo info) throws InterruptedException {
       }
    }
 
+   @org.junit.jupiter.api.Test
+   public void testStraightforwardTransformation(TestInfo info) throws InterruptedException {
+      BlockingQueue<DataSet> dataSetQueue = eventConsumerQueue(DataSet.class, DataSet.EVENT_NEW);
+      NamedJsonPath path = createExampleNamedJsonPath("foo", "$.value");
+      Schema schema = createExampleSchema(info);
+      assertNotNull(schema.id);
+      Transformer transformer = createExampleTransformerWithJsonPath("acme", schema, path);
+      Test exampleTest = createExampleTest(getTestName(info));
+      Test test = createTest(exampleTest);
+      overwriteTransformer(test, transformer);
+      uploadRun(schema, test.name);
+
+      DataSet event = dataSetQueue.poll(10, TimeUnit.SECONDS);
+      assertNotNull(event);
+      JsonNode node = event.data;
+      assertEquals("[42]", node.asText());
+   }
+
+   private NamedJsonPath createExampleNamedJsonPath(String name, String path) {
+      NamedJsonPath jsonpath = new NamedJsonPath();
+      jsonpath.name = name;
+      jsonpath.jsonpath = path;
+      return jsonpath;
+   }
+
+   @org.junit.jupiter.api.Test
+   public void testTransformationWithoutSchema(TestInfo info) throws InterruptedException {
+      BlockingQueue<DataSet> dataSetQueue = eventConsumerQueue(DataSet.class, DataSet.EVENT_NEW);
+      Test test = createTest(createExampleTest(getTestName(info)));
+      setTestVariables(test, "Value", "value");
+
+      ObjectNode jsonWithoutSchema = JsonNodeFactory.instance.objectNode()
+         .put("meaning of life", "42");
+      uploadRun(jsonWithoutSchema.toString(), test.name);
+
+      try (CloseMe ignored = roleManager.withRoles(em, Arrays.asList(TESTER_ROLES))) {
+         DataSet event = dataSetQueue.poll(10, TimeUnit.SECONDS);
+         assertNotNull(event);
+      }
```

Comment:
let's assert that result is `[]` as well.

Reply:
yes the tests have been cleaned up.